### PR TITLE
Fix #97 Upgrade the Back with JH4

### LINKED ISSUE #97: Upgrade the back-end to JHipster 4.x
Currently the registry is generated with JHipster 3.x, it should be upgraded to JHipster 4.x

### PR DESCRIPTION
**Pull request by JulienMrgrd**
Fix #97 with JH 4.0.6

### CODE REVIEW

**Comment by JulienMrgrd** [comment]
I'm writing some JUnit tests.

File: src/main/docker/app.yml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+version: '2'
```
Comment (PierreBesson) [inline src/main/docker/app.yml]: As this is the registry itself, this file must be changed as it doesn't need to register itself to another registy.

File: src/main/java/io/github/jhipster/registry/web/rest/util/HeaderUtil.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
+package io.github.jhipster.registry.web.rest.util;
```
Comment (PierreBesson) [inline src/main/java/io/github/jhipster/registry/web/rest/util/HeaderUtil.java]: Not needed as there are no entities on the registry.

File: src/main/java/io/github/jhipster/registry/web/rest/util/PaginationUtil.java [hunk 3]
```diff
@@ -0,0 +1,48 @@
+package io.github.jhipster.registry.web.rest.util;
```
Comment (PierreBesson) [inline src/main/java/io/github/jhipster/registry/web/rest/util/PaginationUtil.java]: Not needed as there are no entities on the registry.

File: src/main/resources/i18n/messages.properties [hunk 4]
```diff
@@ -0,0 +1,22 @@
+# Error page
```
Comment (PierreBesson) [inline src/main/resources/i18n/messages.properties]: Please also remove all internationalization files.

Comment (JulienMrgrd) [inline src/main/resources/i18n/messages.properties]: Ok. Should I also delete those ? :

- the dir 
- addInterceptors in java.io...config.LocaleConfiguration.java
- ".antMatchers("/i18n/**")" in MicroserviceSecurityConfiguration.java

File: src/main/docker/app.yml (hunk 1 above)
Comment (JulienMrgrd) [inline src/main/docker/app.yml]: Among all your modification messages, it's the only that I'm not sure to know how to resolve.

Comment (pascalgrimaud) [inline src/main/docker/app.yml]: I think, we can delete this file, it is no need (see the folder https://github.com/jhipster/jhipster-registry/tree/master/src/main/docker)
Do you confirm @PierreBesson ?

File: src/main/java/io/github/jhipster/registry/config/ThymeleafConfiguration.java [hunk 5]
```diff
@@ -0,0 +1,28 @@
+package io.github.jhipster.registry.config;
```
Comment (JulienMrgrd) [inline src/main/java/io/github/jhipster/registry/config/ThymeleafConfiguration.java]: Better than JSP ? :o
Ok, it's deleted and also : 
- "thymeleaf.mode: XHTML" in all application*.yml
- the line "<logger name="org.thymeleaf" level="WARN"/>" in logback-spring.xml

File: README.md [hunk 6]
```diff
@@ -1,21 +1,75 @@
-# JHipster Registry
+# jhipsterRegistry
+This application was generated using JHipster 4.0.6, you can find documentation and help at [https://jhipster.github.io/documentation-archive/v4.0.6](https://jhipster.github.io/documentation-archive/v4.0.6).
 
-[![Build Status][travis-image]][travis-url]
+This is a "microservice" application intended to be part of a microservice architecture, please refer to the [Doing microservices with JHipster][] page of the documentation for more information.
```
Comment (JulienMrgrd) [inline README.md]: Oups, my bad.

File: src/main/docker/app.yml (hunk 1 above)
Comment (PierreBesson) [inline src/main/docker/app.yml]: yes you can remove this.

File: src/main/resources/i18n/messages.properties (hunk 4 above)
Comment (PierreBesson) [inline src/main/resources/i18n/messages.properties]: Yes if it doesn't break anything.

File: src/main/docker/app.yml (hunk 1 above)
Comment (JulienMrgrd) [inline src/main/docker/app.yml]: Ok

File: src/main/java/io/github/jhipster/registry/config/MicroserviceSecurityConfiguration.java [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
+import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;
+import org.springframework.security.config.http.SessionCreationPolicy;
+import org.springframework.security.data.repository.query.SecurityEvaluationContextExtension;
+
+@Configuration
+@EnableWebSecurity
+@EnableGlobalMethodSecurity(prePostEnabled = true, securedEnabled = true)
+@Order(10)
+public class MicroserviceSecurityConfiguration extends WebSecurityConfigurerAdapter {
```
Comment (jdubois) [inline src/main/java/io/github/jhipster/registry/config/MicroserviceSecurityConfiguration.java]: Check with `SecurityConfiguration`, you should only have one of them

File: src/main/java/io/github/jhipster/registry/domain/package-info.java [hunk 8]
```diff
@@ -0,0 +1,4 @@
+/**
+ * JPA domain objects.
+ */
+package io.github.jhipster.registry.domain;
```
Comment (jdubois) [inline src/main/java/io/github/jhipster/registry/domain/package-info.java]: you should delete unneeded packages

Comment (JulienMrgrd) [inline src/main/java/io/github/jhipster/registry/domain/package-info.java]: Packages deleted in the #3356b26 commit

File: src/main/java/io/github/jhipster/registry/config/MicroserviceSecurityConfiguration.java (hunk 7 above)
Comment (JulienMrgrd) [inline src/main/java/io/github/jhipster/registry/config/MicroserviceSecurityConfiguration.java]: Done in commit #3abf570

**Comment by JulienMrgrd** [comment]
Do you agree with the .travis.yml modifications to automate the junit tests at each commit ?
(N.B : I commit tests progressively)

**Comment by PierreBesson** [comment]
@JulienMrgrd, @jdubois, if we do the tests here, shouldn't we add `-DskipTests`in the docker build so that we don't run the tests two times ?

**Comment by JulienMrgrd** [comment]
Yes, indeed, I haven't seen the tests inside the Docker command on Travis.

**Comment by jdubois** [comment]
Great, let's do this!

**Comment by JulienMrgrd** [comment]
About **tests**, is missing :

- LogginAspect.java
- the config package
- Http401UnauthorizedEntryPoint.java
- EurekaRessource.java
- ApplicationWebXml.java
- JhipsterRegistryApp.java

All controllers, VM (old "DTO"), and class in the "errors" package are done.

However, I've to work on the **[Zuul support task](https://github.com/jhipster/jhipster-registry/issues/99)**. So I stop to produce tests at this time, and so if it's OK for you, you can merge the PR. :+1:

**Comment by jdubois** [comment]
Thanks @JulienMrgrd I just merged it! Yes the Zuul part is far more important now.

**Comment by pascalgrimaud** [comment]
a lot of work here! nice @JulienMrgrd 
About the missing tests, I'm not sure we have them in the main generator-jhipster, it can be done one day, so we can improve the Sonar coverage ;-)

**Pull request merged**